elastalert/alerters/httppost2.py

```diff
@@ -30,6 +43,7 @@ def __init__(self, rule):
     def alert(self, matches):
         """ Each match will trigger a POST to the specified endpoint(s). """
         for match in matches:
+            match = _escape_all_values(match)
```

Comment:
Thanks for the contribution!

To avoid breaking existing users that do not want the match values encoded, particularly those that are using the `post_all_values`, `post_raw_fields`, etc config options, please adjust the PR to only send the encoded data into the `payload_template.render()` function.

Please add unit tests for the changed functionality.

Reply:
done